### PR TITLE
bpf: minor LB cleanups

### PR DESCRIPTION
**Pull request by julianwiedmann**
A few small LB cleanups that have accumulated. Ideally the first ~~two~~ three will make review of a pending fix a bit easier.

### CODE REVIEW

File: bpf/lib/lb.h [hunk 1]
```diff
@@ -1033,18 +1032,21 @@ lb6_to_lb4_service(const struct lb6_service *svc __maybe_unused)
...
 					 const struct ct_state *ct_state, bool has_l4_header)
 {
+	struct csum_offset csum_off = {};
 	__be32 old_sip, new_sip, sum = 0;
 	int ret;
 
 	cilium_dbg_lb(ctx, DBG_LB4_REVERSE_NAT, nat->address, nat->port);
 
+	if (has_l4_header)
+		csum_l4_offset_and_flags(tuple->nexthdr, &csum_off);
```
Comment (gentoo-root) [inline bpf/lib/lb.h:1046]: Offtopic question: why don't we set `off->offset` for IPPROTO_ICMP in csum_l4_offset_and_flags?

**Review by gentoo-root — APPROVED** [review]
Looks good to me.

File: bpf/lib/lb.h (hunk 1 above)
Comment (julianwiedmann) [reply bpf/lib/lb.h:1046]: I've wondered about that myself recently while poking around the NAT code :) (as it requires us to set the offset manually ...). Best I could come up with was that we didn't need it in the pure LB paths.

So one option would be to declare the existing helpers as LB-specific, and introduce generic variants that *do* handle ICMP.

**Pull request merged**